app/src/main/java/io/scalac/degree/android/fragment/SpeakersFragment.java

```diff
@@ -103,39 +126,95 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
     }
 
     private void populateList() {
-        final Realm realm = realmProvider.getRealm();
-        final RealmResults<RealmSpeaker> realmList = realm.allObjects(RealmSpeaker.class);
+        final List<RealmSpeakerShort> speakers =
+                speakersDataManager.getAllShortSpeakers();
 
-        final List<RealmSpeaker> finalResult = Stream.of(realmList).
-                sortBy(new Function<RealmSpeaker, Comparable>() {
+        final List<SpeakersGroup> list = Stream.of(speakers)
+                .groupBy(new Function<RealmSpeakerShort, String>() {
                     @Override
-                    public Comparable apply(RealmSpeaker value) {
-                        return value.getLastName();
+                    public String apply(RealmSpeakerShort value) {
+                        return value.getFirstName().substring(0, 1);
                     }
-                }).collect(Collectors.<RealmSpeaker>toList());
+                })
+                .map(new Function<Map.Entry<String, List<RealmSpeakerShort>>, SpeakersGroup>() {
+                    @Override
+                    public SpeakersGroup apply(Map.Entry<String, List<RealmSpeakerShort>> value) {
+                        final List<RealmSpeakerShort> list =
+                                Stream.of(value.getValue())
+                                        .sortBy(new Function<RealmSpeakerShort, Comparable>() {
+                                            @Override
+                                            public Comparable apply(RealmSpeakerShort value) {
+                                                return value.getFirstName();
+                                            }
+                                        })
+                                        .collect(Collectors.<RealmSpeakerShort>toList());
+                        return new SpeakersGroup(value.getKey(), list);
+                    }
+                })
+                .sortBy(new Function<SpeakersGroup, Comparable>() {
+                    @Override
+                    public Comparable apply(SpeakersGroup value) {
+                        return value.getGroupLetter();
+                    }
+                })
+                .collect(Collectors.<SpeakersGroup>toList());
 
-        Logger.l("Speakers to show: " + finalResult.size());
+        itemAdapter.setSpeakers(list);
+        itemAdapter.notifyDataSetChanged();
+    }
 
-        itemAdapter = new ItemAdapter(finalResult);
-        listView.setAdapter(itemAdapter);
+    private Runnable showProgressRunnable = new Runnable() {
+        @Override
+        public void run() {
+            progressBar.setVisibility(View.VISIBLE);
+        }
+    };
+
+    private void showProgress() {
```

Comment:
Ale chce mieć uchwyt do tego, żeby później móc to usunąć z handlera jeśli hide będzie wywołany przed show.